sdk/storage/Azure.Storage.Queues/tests/QueueClientTests.cs

```diff
@@ -1977,5 +1977,49 @@ public void CanMockClientConstructors()
             mock = new Mock<QueueClient>(new Uri("https://test/test"), new AzureSasCredential("foo"), new QueueClientOptions()).Object;
             mock = new Mock<QueueClient>(new Uri("https://test/test"), mockTokenCredential, new QueueClientOptions()).Object;
         }
+
+        [RecordedTest]
+        [TestCase("")]
+        [TestCase("ra")]
+        [TestCase("raup")]
+        [TestCase("paru")]
+        [TestCase("zaxpqr")]
+        public void QueueClientPolicyPermissions_checkIfSetCorrectly(string permissions)
+        {
+            QueueAccessPolicy qap = new QueueAccessPolicy
+            {
+                Permissions = permissions
+            };
+
+            string permissionsStr = "";
+            QueueAccessPolicyPermissions permissionsEnum = QueueAccessPolicyPermissions.None;
+
+            switch (permissions)
+            {
+                case "":
+                    permissionsStr = "";
+                    permissionsEnum = QueueAccessPolicyPermissions.None;
+                    break;
+                case "ra":
+                    permissionsStr = "ra";
+                    permissionsEnum = QueueAccessPolicyPermissions.Read | QueueAccessPolicyPermissions.Add;
+                    break;
+                case "raup":
+                    permissionsStr = "raup";
+                    permissionsEnum = QueueAccessPolicyPermissions.All;
+                    break;
+                case "paru":
+                    permissionsStr = "raup";
+                    permissionsEnum = QueueAccessPolicyPermissions.All;
+                    break;
+                case "zaxpqr":
```

Comment:
I find this test case a little concerning @seanmcc-msft should we be dropping invalid Permissions if the customer hands us invalid permissions?

Reply:
Yeah I called out about this in the throw exception comment. We shouldn't have a case like this where the permission letters are not valid (especially since the permission used to be in string form without validation) but if we don't do validations I think this test should be the expected outcome in my opinion (please let me know otherwise).

I also don't know if its possible to have invalid permissions based on whats "upstream" but curious to hear yall's thoughts.